src/Consumer/UserAccountConsumer.php

```diff
@@ -16,33 +17,47 @@ class UserAccountConsumer implements ConsumerInterface
     private $repository;
     private $logger;
     private $membershipRequestHandler;
+    private $retryProducer;
 
     public function __construct(
         AdherentAccountProvider $provider,
         AdherentRepository $repository,
         MembershipRequestHandler $membershipRequestHandler,
-        LoggerInterface $logger
+        LoggerInterface $logger,
+        ProducerInterface $retryProducer
     ) {
         $this->provider = $provider;
         $this->repository = $repository;
         $this->membershipRequestHandler = $membershipRequestHandler;
         $this->logger = $logger;
+        $this->retryProducer = $retryProducer;
     }
 
     public function execute(AMQPMessage $message): bool
     {
-        if (false === $data = \GuzzleHttp\json_decode($message->getBody(), true)) {
-            return false;
-        }
+        try {
+            if (false === $data = \GuzzleHttp\json_decode($message->getBody(), true)) {
+                throw new \RuntimeException(sprintf('cannot decode the message. Body: "%s"', $message->getBody()));
+            }
+
+            switch ($key = $message->get('routing_key')) {
+                case 'user.modification':
+                    return $this->updateUser($key, $data);
+                case 'user.deletion':
+                    return $this->deleteUser($key, $data);
+            }
+
+            throw new \RuntimeException(sprintf('Routing key "%s" cannot be handled', $key));
+        } catch (\Throwable $e) {
```

Comment:
avec ce catch on risque de re traiter les cas:
 - cannot decode the message.
 - Routing key "%s" cannot be handled
et ces messages vont jamais disparaître. 

si le process consumer s'arrête avec une exception, alors le message reste dans la queue il me semble, pourquoi re développer ce comportement?

Reply:
Nope en faite on redirige sur un autre exchange qui lui ajoute le message dans une queue qui attend 1h avant de remettre les messages dans cette queue.

Reply:
mais au final le message qui déclenche l'exception `cannot decode the message.` il va se balader d'une queue à l'autre et sera jamais supprimé non?

Reply:
Le but c'est de remonter les erreurs dans Sentry et donc d'analyser le problème et si il faut supprimer le message.

Reply:
👍